Doc/library/ipaddress.rst

```diff
@@ -377,14 +379,17 @@ so to avoid duplication they are only documented for :class:`IPv4Network`.
       a slash (``/``).  The IP address is the network address, and the mask
       can be either a single number, which means it's a *prefix*, or a string
       representation of an IPv4 address.  If it's the latter, the mask is
-      interpreted as a *net mask* if it starts with a non-zero field, or as
-      a *host mask* if it starts with a zero field.  If no mask is provided,
-      it's considered to be ``/32``.
+      interpreted as a *net mask* if possible, or as a *host mask* otherwise.
```

Comment:
I don't like the change here. It's confusing. What is `if possible` mean? With the below note, the original doc is fine. Or refer to the comment of `IPv4Network.__init__`:

>

Reply:
"starts with a zero field" is wrong.  128.0.0.0 is* accepted as the netmask for a /1 network, 127.255.255.255 is* is accepted as the hostmask for a /1 network, neither starts with a zero field.

How about: "interpreted as a *net mask* if it is a valid *net mask*, or interpreted as a *host mask* otherwise."?

(* Or they were when I originally wrote the patch, I haven't looked to see if the code has changed since).

Reply:
Sorry, Jon, I pushed my changes while you were leaving this comment.  I will make another commit once the proper wording is worked out.

Reply:
@jonfoster Thanks for your reply! But I always understand the non-zero field means the binary zero not octet zero. So 128.0.0.0 is starting with non-zero field and should be netmask. And 127.255.255.255 is starting with zero field and should be hostmask.